Makefile

```diff
@@ -21,35 +21,49 @@ ifeq (run,$(firstword $(MAKECMDGOALS)))
 endif
 
 .PHONY: all
+## Alias for `build`
 all: build
 
 .PHONY: build
+## Build runtime docker image
 build:
 	./scripts/build_docker.sh --target runtime
 
 .PHONY: build.develop
+## Build develop docker image
 build.develop:
 	./scripts/build_docker.sh --target develop
 
 .PHONY: develop
+## Build develop docker image and run an interactive shell in the develop envionment
 develop: build.develop
 	./scripts/develop.sh
 
 .PHONY: run
+## Build develop docker image and run a make command in the develop envionment (e.g. `make run fmt` will execute `make fmt` within the docker container)
 run: build.develop
 	./scripts/develop.sh make $(RUN_ARGS)
 
 .PHONY: test
+## Run tests
 test:
 	./scripts/run_tests.sh
 
 .PHONY: fmt
+## Run formatting
 fmt:
 	./scripts/fmt.sh
 
 .PHONY: proto.compile
+## Compile protos
 proto.compile:
 	./scripts/compile_protos.sh
 
+.PHONY: help
+## Print Makefile documentation
+help:
+	@perl -0 -nle 'printf("%-25s - %s\n", "$$2", "$$1") while m/^##\s*([^\r\n]+)\n^([\w-]+):[^=]/gm' $(MAKEFILE_LIST) | sort
```

Comment:
the regex needs to include `.` (dot) to print help for targets like `proto.compile` or `build.develop`

`'printf("%-25s - %s\n", "$$2", "$$1") while m/^##\s*([^\r\n]+)\n^([\w-]+):[^=]/gm'`    :x: 
`'printf("%-25s - %s\n", "$$2", "$$1") while m/^##\s*([^\r\n]+)\n^([\w.-]+):[^=]/gm'`  ✅ 
`                                                                      ^             `

Reply:
```suggestion
	@perl -0 -nle 'printf("\033[36m  %-15s\033[0m %s\n", "$$2", "$$1") while m/^##\s*([^\r\n]+)\n^([\w.-]+):[^=]/gm' $(MAKEFILE_LIST) | sort
```